x-pack/plugins/alerting/server/routes/snooze_rule.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { ILicenseState, AlertTypeDisabledError } from '../lib';
+import { verifyAccessAndContext } from './lib';
+import { AlertingRequestHandlerContext, INTERNAL_BASE_ALERTING_API_PATH } from '../types';
+
+const paramSchema = schema.object({
+  id: schema.string(),
+});
+
+const bodySchema = schema.object({
+  snooze_end_time: schema.oneOf([
```

Comment:
Yes you are right, but we do not have all the requirements yet.  It will evolve for sure and that's why your idea to keep the now and the duration might help us with migration in the future. 
That's also why we are keeping the `_snooze` API internal and hiding the `snooze_end_time` from our public API.

Reply:
resolve with https://github.com/elastic/kibana/pull/127081#issuecomment-1067274198